### PR TITLE
RPATH on by default

### PR DESCRIPTION
**Pull request by branfosj**
Documentation for https://github.com/easybuilders/easybuild-framework/pull/4448

### CODE REVIEW

File: docs/easybuild-v5/enhancements.md [hunk 1]
```diff
@@ -13,6 +14,28 @@ Various significant enhancements are included in EasyBuild v5.0, including:
...
+
+This enhancement **does not** add any filtering of environment variables. This means that environment variables, such as
+`$LD_LIBRARY_PATH`, will continue to appear in the module files unless EasyBuild is configured to filter these variables
+(see [Relation to `$LD_LIBRARY_PATH`][rpath_support_LD_LIBRARY_PATH]).
+
+To disable RPATH output, either:
+
+* Use the `--disable-rpath` command line option;
+* Set the `$EASYBUILD_DISABLE_RPATH` environment variable;
+* Disable trace mode in a [configuration file](../configuration.md#configuration_file):
```
Comment (boegel) [inline docs/easybuild-v5/enhancements.md:31]:
```suggestion
* Disable RPATH linking in an EasyBuild [configuration file](../configuration.md#configuration_file):
```

File: docs/easybuild-v5/enhancements.md [hunk 2]
```diff
@@ -13,6 +14,28 @@ Various significant enhancements are included in EasyBuild v5.0, including:
...
+
+    ``` ini
+    [override]
+    rapth=0
```
Comment (boegel) [inline docs/easybuild-v5/enhancements.md:35]: typo:

```suggestion
    rpath=0
```

File: docs/easybuild-v5/enhancements.md [hunk 3]
```diff
@@ -13,6 +14,28 @@ Various significant enhancements are included in EasyBuild v5.0, including:
...
+
+## Enable `--rpath` by default {: #rpath }
+
+The [`--rpath` option][rpath_support] is enabled by default (see [easybuild-framework PR #4448](https://github.com/easybuilders/easybuild-framework/pull/4448)). The benefits for enabling RPATH are explained in [Why RPATH?][rpath_support_why].
+
+This enhancement **does not** add any filtering of environment variables. This means that environment variables, such as
+`$LD_LIBRARY_PATH`, will continue to appear in the module files unless EasyBuild is configured to filter these variables
+(see [Relation to `$LD_LIBRARY_PATH`][rpath_support_LD_LIBRARY_PATH]).
+
+To disable RPATH output, either:
```
Comment (boegel) [inline docs/easybuild-v5/enhancements.md:27]:
```suggestion
To disable RPATH linking, either:

File: docs/easybuild-v5/enhancements.md [hunk 4]
```diff
@@ -13,6 +14,28 @@ Various significant enhancements are included in EasyBuild v5.0, including:
...
 
+---
+
+## Enable `--rpath` by default {: #rpath }
+
+The [`--rpath` option][rpath_support] is enabled by default (see [easybuild-framework PR #4448](https://github.com/easybuilders/easybuild-framework/pull/4448)). The benefits for enabling RPATH are explained in [Why RPATH?][rpath_support_why].
+
+This enhancement **does not** add any filtering of environment variables. This means that environment variables, such as
+`$LD_LIBRARY_PATH`, will continue to appear in the module files unless EasyBuild is configured to filter these variables
+(see [Relation to `$LD_LIBRARY_PATH`][rpath_support_LD_LIBRARY_PATH]).
```
Comment (boegel) [inline docs/easybuild-v5/enhancements.md:25]:
```suggestion
This enhancement **does not** add any filtering of environment variables. This means `$LD_LIBRARY_PATH`
will continue to be appended by the environment module files EasyBuild generates,
unless it is configured to filter these variables (via `--filter-env-vars`).
See also [Relation to `$LD_LIBRARY_PATH`][rpath_support_LD_LIBRARY_PATH]).
```

File: docs/easybuild-v5/enhancements.md [hunk 5]
```diff
@@ -13,6 +14,28 @@ Various significant enhancements are included in EasyBuild v5.0, including:
 
 See dedicated page on the new [`run_shell_cmd` function](run_shell_cmd.md).
 
+---
+
+## Enable `--rpath` by default {: #rpath }
+
+The [`--rpath` option][rpath_support] is enabled by default (see [easybuild-framework PR #4448](https://github.com/easybuilders/easybuild-framework/pull/4448)). The benefits for enabling RPATH are explained in [Why RPATH?][rpath_support_why].
```
Comment (boegel) [inline docs/easybuild-v5/enhancements.md:21]:
```suggestion
The [RPATH linking][rpath_support] is enabled by default in EasyBuild v5.0 (see [easybuild-framework PR #4448](https://github.com/easybuilders/easybuild-framework/pull/4448)).

The benefits for enabling RPATH are explained in [Why RPATH?][rpath_support_why].

File: docs/easybuild-v5/enhancements.md [hunk 6]
```diff
@@ -13,6 +14,28 @@ Various significant enhancements are included in EasyBuild v5.0, including:
 
 See dedicated page on the new [`run_shell_cmd` function](run_shell_cmd.md).
 
+---
+
+## Enable `--rpath` by default {: #rpath }
```
Comment (boegel) [inline docs/easybuild-v5/enhancements.md:19]:
```suggestion
## Enable RPATH linking by default {: #rpath }
```

File: docs/easybuild-v5/enhancements.md [hunk 7]
```diff
@@ -5,6 +5,7 @@
 Various significant enhancements are included in EasyBuild v5.0, including:
 
 * [`run_shell_cmd` function][run_shell_cmd]
+* [Enable `--rpath` by default][rpath]
```
Comment (boegel) [inline docs/easybuild-v5/enhancements.md:8]:
```suggestion
* [Enable RPATH linking by default][rpath]
```

**Review by boegel — CHANGES_REQUESTED** [review]

File: docs/easybuild-v5/enhancements.md [hunk 8]
```diff
@@ -13,6 +14,31 @@ Various significant enhancements are included in EasyBuild v5.0, including:
 
 See dedicated page on the new [`run_shell_cmd` function](run_shell_cmd.md).
 
+---
+
+## Enable RPATH linking by default {: #rpath }
+
+The [RPATH linking][rpath_support] is enabled by default in EasyBuild v5.0 (see [easybuild-framework PR #4448](https://github.com/easybuilders/easybuild-framework/pull/4448)).
```
Comment (jfgrimm) [inline docs/easybuild-v5/enhancements.md:21]:
```suggestion
[RPATH linking][rpath_support] is enabled by default in EasyBuild v5.0 (see [easybuild-framework PR #4448](https://github.com/easybuilders/easybuild-framework/pull/4448)).
```

**Review by jfgrimm — APPROVED** [review]
lgtm

File: docs/easybuild-v5/enhancements.md [hunk 9]
```diff
@@ -16,23 +16,26 @@ See dedicated page on the new [`run_shell_cmd` function](run_shell_cmd.md).
...
-This enhancement **does not** add any filtering of environment variables. This means that environment variables, such as
-`$LD_LIBRARY_PATH`, will continue to appear in the module files unless EasyBuild is configured to filter these variables
-(see [Relation to `$LD_LIBRARY_PATH`][rpath_support_LD_LIBRARY_PATH]).
+The benefits for enabling RPATH are explained in [Why RPATH?][rpath_support_why].
 
-To disable RPATH output, either:
+This enhancement **does not** add any filtering of environment variables. This means `$LD_LIBRARY_PATH`
+will continue to be appended by the environment module files EasyBuild generates,
+unless it is configured to filter these variables (via `--filter-env-vars`).
+See also [Relation to `$LD_LIBRARY_PATH`][rpath_support_LD_LIBRARY_PATH]).
```
Comment (boegel) [inline docs/easybuild-v5/enhancements.md:28]:
```suggestion
unless it is configured to filter these variables (via `--filter-env-vars`,
see also [Relation to `$LD_LIBRARY_PATH`][rpath_support_LD_LIBRARY_PATH]).
```

**Review by boegel — CHANGES_REQUESTED** [review]

**Review by boegel — APPROVED** [review]

**Pull request merged**